### ISSUE TITLE
SSL Client

### ISSUE THREAD

**Issue by vishalkothari** [internetofhomethings/Configurable-Web-Server#1]
Hello,
I was scanning the code. For using SSL (WifiClientSecure), isn't the code to load the certificates and verify fingerprints needed for security?

Regards,
Vishal Kothari

**Comment by internetofhomethings** [comment]
The certificate is installed and maintained on the server, not the client. This is the same as when you visit a secure website (https). Using the WifiClientSecure library ensures the esp8266 uses the secure port (8883) when connecting to the mqtt broker. The connection negotiation and traffic are secure, using TSL/SSL encryption

This post  provides instructions for the installation of a certificate on the server:

https://internetofhomethings.com/homethings/?p=1756

**Comment by vishalkothari** [comment]
Thank you. Is there anyway we can use Client certificates for encryption and authentication. I am planning to use AWS IoT MQTT Broker which only supports certificate based auth for MQTT.

**Comment by internetofhomethings** [comment]
I successfully interfaced an ESP8266 with AWS in 2015. This was before the AWS IoT was offered. But after the first year with AWS, they started charging a monthly account maintenance fee whether AWS was used or not. At that point, I opted out of AWS and have never had a need for their services since that time.

I am sure there is a method to accomplish what you want to do. But personally, I do not have any experience using AWS IoT and do not have any desire to pursue it. I prefer to maintain total control by hosting my own MQTT broker on my personal VPS. 

I use https://letsencrypt.org/ certificates, which are free and automatically renewed using a CRON script on my Linux server.

Best wishes to your success in your pursuits!
